sql/analyzer/analyzer.go

```diff
@@ -451,7 +452,15 @@ func newInsertSourceSelector(sel RuleSelector) RuleSelector {
 // Analyze applies the transformation rules to the node given. In the case of an error, the last successfully
 // transformed node is returned along with the error.
 func (a *Analyzer) Analyze(ctx *sql.Context, n sql.Node, scope *Scope) (sql.Node, error) {
-	n, _, err := a.analyzeWithSelector(ctx, n, scope, SelectAllBatches, DefaultRuleSelector)
+	value, err := ctx.GetSessionVariable(ctx, "debug_analyzer")
```

Comment:
This works for debugging, but in production we need to avoid impacting other sessions. Everywhere we switch on `a.Verbose && a.Debug` would want to be updated to `(a.Verbose && a.Debug) || debugAnalyzer`, with the value always drawn from the current session.

Reply:
I think the analyzer is a singleton